docs/batch jobs/ISEL timetable extraction batch job/isel-timetable-batch-specification.md

```diff
@@ -0,0 +1,213 @@
+# ISEL Timetable Batch Specification
+
+## Overview
+
+This documents aims to specify the process of extracting information from the timetable document published by ISEL in the beggining of each semester. For more information on how the isel timetable extraction batch job uses the components of Spring Batch, refer to the [batch job architecture page]().
+
+## Extractable information
+
+From the timetable document there is a wide variety of information that can be extracted about school activities and organization, including course offer and the teaching staff of the course. We divide the information according to its source, as it is found on the page header or on the tables in the center and footer of the page. This categorization is relevant in the context of the batch job, because the library used to parse the section of the page that contains tabular data (tabula) is not the same that is used to parse the headers (iText) and so these two segments of the job are independent.
+
+Following we have the information that can be extracted from the timetable document grouped in two clusters of independent data, timetable and faculty, in json format.
+### Timetable
+
+```json
+{
```

Comment:
This description language is not immediately clear to me. Perhaps a concrete example, followed by the schema would be easier to understand. Another option would be to describe this model using Kotlin data classes.
What is the purpose of the `title` in this definition language?

Reply:
Json excerpts replaced with actual examples. Data classes included in the specification doc.